typescript-methods/main.ts

```diff
@@ -0,0 +1,97 @@
+// Create a new array containing four strings that are
+// the names of your favorite superheroes and assign it to a new variable named heroes.
+const heroes: string[] = ['Spider-Man', 'Iron Man', 'Black Widow', 'Thor'];
+console.log(heroes);
+// Call the random() method of the Math object and
+// assign its return value to a new variable named randomNumber.
+let randomNumber: number = Math.random();
+console.log('randomNumber before:', randomNumber);
+// Multiply your random number by the length property of your heroes object
+// and assign the result back to the randomNumber variable.
+randomNumber = randomNumber * heroes.length;
+console.log('randomNumber after:', randomNumber);
+// Call the floor() method of the Math object and pass in randomNumber as an argument.
+// Assign its return value to a new variable named randomIndex.
+const randomIndex: number = Math.floor(randomNumber);
+console.log('randomIndex:', randomIndex);
+// Using bracket notation, access the heroes array at randomIndex
+// and assign the result to a new variable named randomHero.
+const randomHero: string = heroes[randomIndex];
+// Log the value of randomHero to the console.
+console.log('Random Hero:', randomHero);
+// Create a new array containing three objects.
+const library: Array<{ title: string; author: string }> = [
+  { title: 'The Great Gatsby', author: 'F. Scott Fitzgerald' },
+  { title: 'To Kill a Mockingbird', author: 'Harper Lee' },
+  { title: '1984', author: 'George Orwell' },
+];
```

Comment:
Totally valid! I probably would have created an interface instead since you plan on reusing this same type in multiple places
```suggestion
interface Book {
  title: string;
  author: string;
}

const library: Array<Book> = [
  { title: 'The Great Gatsby', author: 'F. Scott Fitzgerald' },
  { title: 'To Kill a Mockingbird', author: 'Harper Lee' },
  { title: '1984', author: 'George Orwell' },
];
```